equinox/nn/pool.py

```diff
@@ -0,0 +1,201 @@
+import jax.numpy as jnp
+import numpy as np
+
+from ..custom_types import Array
+from ..module import Module, static_field
+
+
+class Pool(Module):
+    operation: Callable[[Array], Array] = static_field()
+    num_spatial_dims: int = static_field()
+    kernel_size: Union[int, Sequence[int]] = static_field()
+    strides: Sequence[int] = static_field()
+    padding: Union[str, Sequence[Tuple[int, int]]] = static_field()
+    init: float = static_field()
+
+    def __init__(
+        self,
+        operation: Callable[[Array], Array],
+        num_spatial_dims: int,
+        kernel_size: Union[int, Sequence[int]],
+        strides: Sequence[int] = None,
+        padding: Union[str, Sequence[Tuple[int, int]]] = "VALID",
+        **kwargs,
+    ):
+        """**Arguments:**
+        - `operation`: The operation applied to the inputs of each window.
+        - `num_spatial_dims`: The number of spatial dimensions.
+        - `kernel_size`: The size of the convolutional kernel.
+        - `stride`: The stride of the convolution.
+        - `padding`: The amount of padding to apply before and after each
+            spatial dimension or a string specifying `VALID` or `SAME` (SAME
+            only applicable when stride is (1, 1)).
+        """
+        super().__init__(**kwargs)
+
+        self.operation = operation
+        self.init = 0.0
+        if operation is lax.max:
+            self.init = -jnp.inf
+        self.num_spatial_dims = num_spatial_dims
+        self.kernel_size = kernel_size
+        self.strides = strides
+        if strides is None:
+            self.strides = (1,) * len(kernel_size)
+        self.padding = padding
+
+    def __call__(
+        self, x: Array, *, key: Optional["jax.random.PRNGKey"] = None
+    ) -> Array:
+        """**Arguments:**
+        - `x`: The input. Should be a JAX array of shape `(in_channels, dim_1, ..., dim_N)`, where
+            `N = num_spatial_dims`.
+        - `key`: Ignored; provided for compatibility with the rest of the Equinox API.
+            (Keyword only argument.)
+        **Returns:**
+        A JAX array of shape `(in_channels, new_dim_1, ..., new_dim_N)`.
+        """
+        assert len(x.shape) == self.num_spatial_dims + 1, \
+            (f'Input should have {self.num_spatial_dims} spatial dimensions, '
+             f'but input has shape {x.shape}')
+
+        x = einops.rearrange(x, 'a ... -> ... a')
+        x = jnp.expand_dims(x, axis=0)
+        x = lax.reduce_window(
+            x,
+            self.init,
+            self.operation,
+            (1,) + self.kernel_size + (1,),
+            (1,) + self.strides + (1,),
+            self.padding,
+        )
+        if self.operation == lax.add:
+            x = x / np.prod(self.kernel_size)
+
+        x = jnp.squeeze(x, axis=0)
+        x = einops.rearrange(x, '... a -> a ...')
+        return x
+
+
+class AvgPool1D(Pool):
+    """As [`equinox.nn.Pool`][] with `operation=lax.add` and
+    `num_spatial_dims=1`."""
```

Comment:
I can tell you've copied the style of the convolutional layers here, but I think this docstring is a bit opaque. Something like "downsamples by taking an average over a sliding window" might be a helpful one-sentence summary.